core/src/main/java/edu/wpi/grip/core/operations/composite/SimpleFilterContoursOperation.java

```diff
@@ -0,0 +1,113 @@
+package edu.wpi.grip.core.operations.composite;
+
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.OperationDescription;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sockets.SocketHint;
+import edu.wpi.grip.core.sockets.SocketHints;
+import edu.wpi.grip.core.util.Icon;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static org.bytedeco.javacpp.opencv_core.Mat;
+import static org.bytedeco.javacpp.opencv_core.MatVector;
+import static org.bytedeco.javacpp.opencv_imgproc.arcLength;
+import static org.bytedeco.javacpp.opencv_imgproc.contourArea;
+
+/**
+ * An {@link Operation} that takes in a list of contours and outputs a list of any contours in the
+ * input that match all of several criteria.  The user can specify a minimum area and perimeter.
+ * This is useful because running a FindContours on a real-life image typically leads to many small
+ * undesirable contours from noise and small objects, as well as contours that do not meet the
+ * expected characteristics of the feature we're actually looking for.  So, this operation can
+ * help narrow them down.
+ */
+public class SimpleFilterContoursOperation implements Operation {
+
+  public static final OperationDescription DESCRIPTION =
+      OperationDescription.builder()
+          .name("Simple Filter Contours")
+          .summary("Find contours matching certain criteria")
+          .category(OperationDescription.Category.FEATURE_DETECTION)
+          .icon(Icon.iconStream("find-contours"))
+          .build();
+
+  private final SocketHint<ContoursReport> contoursHint = new SocketHint.Builder<>(ContoursReport
+      .class)
+      .identifier("Contours").initialValueSupplier(ContoursReport::new).build();
+
+  private final SocketHint<Number> minAreaHint =
+      SocketHints.Inputs.createNumberSpinnerSocketHint("Min Area", 0, 0, Integer.MAX_VALUE);
+
+  private final SocketHint<Number> minPerimeterHint =
+      SocketHints.Inputs.createNumberSpinnerSocketHint("Min Perimeter", 0, 0, Integer.MAX_VALUE);
+
+
+  private final InputSocket<ContoursReport> contoursSocket;
+  private final InputSocket<Number> minAreaSocket;
+  private final InputSocket<Number> minPerimeterSocket;
+
+  private final OutputSocket<ContoursReport> outputSocket;
+
+  @SuppressWarnings("JavadocMethod")
+  public SimpleFilterContoursOperation(InputSocket.Factory inputSocketFactory, OutputSocket.Factory
```

Comment:
Put parameters on separate lines to avoid a newline inside a parameter definition